dnf/cli/demand.py

```diff
@@ -58,3 +58,8 @@ class DemandSheet(object):
     changelogs = _BoolDefault(False)
 
     transaction_display = None
+
+    # This demand controlls applicability of the plugins that could filter
+    # repositories packages (e.g. versionlock).
+    # If it stays None, the demands.resolving is used as a fallback.
+    plugin_filtering_enabled = _BoolDefault(None)
```

Comment:
This is real Python bool option - True, False, None. But what about future in C++?

Reply:
The name of the class _BoolDefault is kind of misleading and it can accommodate any variable type.
For compatibility reason I needed three possible values here:
True - filtering is enabled
False - filtering is disabled
NOT SET - use previous mechanism
The other way would be to drop demands.resolving from versionlock plugin and make decision based only on the new demand. I can do it for commands in dnf*, but what about third party plugins?
As far as future is concerned - we will need to rethink the demands anyway. Even now not all of them are boolean values - see transaction_display, success_exit_status.

Reply:
LGTM